homeassistant/components/particle/__init__.py

```diff
@@ -0,0 +1,100 @@
+"""Integration with the Particle Cloud."""
+
+DOMAIN = 'particle'
+
+import logging
+import voluptuous as vol
+from homeassistant.const import (
+  STATE_ON,
+  STATE_UNAVAILABLE,
+)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.config_validation import ENTITY_SERVICE_SCHEMA
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_component import EntityComponent
+
+_LOGGER = logging.getLogger(DOMAIN)
+CONFIG_SCHEMA = vol.Schema({
+  DOMAIN: vol.Schema({
+    vol.Required('access_token'): cv.string,
+  }),
+}, extra=vol.ALLOW_EXTRA)
+
+FUNCTION_CALL_SCHEMA = ENTITY_SERVICE_SCHEMA.extend({
+  vol.Required('function'): cv.string,
+  vol.Optional('args'): vol.All(
+    cv.ensure_list, vol.Length(min=1), [cv.string]
+  ),
+})
+
+def setup(hass, config):
+  """Initialize our component while Home Assistant loads."""
+  from pyparticleio.ParticleCloud import ParticleCloud
+
+  access_token = config[DOMAIN]['access_token']
+
+  particle = hass.data[DOMAIN] = ParticleCloud(access_token)
+
+  component = EntityComponent(_LOGGER, DOMAIN, hass)
+
+  devices = {}
+  device_list = []
+
+  for name, info in particle.devices.items():
```

Comment:
I assume that if the access token is not valid that there will be a traceback. You should check, if possible, that the integration works during the setup process and make it fails if not. There is no way to recover if the credentials are wrong.